optimum/intel/ipex/modeling_base.py

```diff
@@ -414,7 +413,7 @@ def forward(
             position_ids = attention_mask.long().cumsum(-1) - 1
             position_ids.masked_fill_(attention_mask == 0, 1)
             if past_key_values:
-                position_ids = position_ids[:, -1].unsqueeze(-1)
+                position_ids = position_ids[:, -input_ids.shape[-1] :]
```

Comment:
why is this modification needed ?

Reply:
`position_ids` should always have the same size as `input_ids`, we cannot assume the length is `1` while pkv exists (for example, assisted decoding).

Reply:
Shouldn't this be taken care by `prepare_inputs_for_generation`?

Reply:
It's also an option, WDYT @echarlaix ?

Reply:
it's already integrated in `prepare_inputs_for_generation` for each modeling 
https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py#L1237, could make sense to remove it all actually